pkg/services/alerting/notifiers/wecom.go

```diff
@@ -0,0 +1,237 @@
+	}
+
+	msgType := model.Settings.Get("msgType").MustString(weComMsgTypeMarkdown)
+	uploadSingleImage := model.Settings.Get("uploadSingleImage").MustBool()
+
+	return &WeComNotifier{
+		NotifierBase:      NewNotifierBase(model),
+		MsgType:           msgType,
+		URL:               url,
+		UploadSingleImage: uploadSingleImage,
+		log:               log.New("alerting.notifier.wecom"),
+	}, nil
+}
+
+// WeComNotifier is responsible for sending alert notifications to WeCom.
+type WeComNotifier struct {
+	NotifierBase
+	MsgType           string
+	URL               string
+	UploadSingleImage bool
+	log               log.Logger
+}
+
+// Notify send an alert notification to WeCom.
+func (w *WeComNotifier) Notify(evalContext *alerting.EvalContext) error {
+	w.log.Info("Sending WeCom")
+
+	body, err := w.buildBody(evalContext)
+	if err != nil {
+		return err
+	}
+
+	cmd := &models.SendWebhookSync{
+		Url:  w.URL,
+		Body: string(body),
+	}
+
+	if err := bus.DispatchCtx(evalContext.Ctx, cmd); err != nil {
+		w.log.Error("Failed to send WeCom", "error", err, "wecom", w.Name)
+		return err
+	}
+
+	if w.UploadSingleImage && w.NeedsImage() {
+		if err := w.uploadImage(evalContext); err != nil {
+			w.log.Error("Failed to send WeCom image", "error", err)
+		}
+	}
+
+	return nil
+}
+
+func (w *WeComNotifier) uploadImage(evalContext *alerting.EvalContext) error {
+	if _, err := os.Stat(evalContext.ImageOnDiskPath); err != nil {
+		return nil
+	}
+
+	imgFile, err := os.Open(evalContext.ImageOnDiskPath)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = imgFile.Close()
+	}()
+
+	f, err := ioutil.ReadAll(imgFile)
+	if err != nil {
+		return err
+	}
+
+	imgBody := map[string]interface{}{
+		"msgtype": "image",
+		"image": map[string]string{
+			"base64": base64.StdEncoding.EncodeToString(f),
+			"md5":    fmt.Sprintf("%x", md5.Sum(f)),
+		},
+	}
+
+	imgBodyJSON, err := json.Marshal(imgBody)
+	if err != nil {
+		return err
+	}
+
+	imgCmd := &models.SendWebhookSync{
+		Url:  w.URL,
+		Body: string(imgBodyJSON),
+	}
+
+	if err := bus.DispatchCtx(evalContext.Ctx, imgCmd); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w *WeComNotifier) truncate(str string, length int) string {
+	if len(str) <= length {
+		return str
+	}
+	return str[:length-1] + "..."
+}
+
+func (w *WeComNotifier) buildNewsBody(evalContext *alerting.EvalContext) ([]byte, error) {
+	body := map[string]interface{}{
+		"msgtype": w.MsgType,
+	}
+	messageURL, _ := evalContext.GetRuleURL()
+	body["news"] = map[string]interface{}{
+		"articles": []map[string]interface{}{
+			{
+				"title":       evalContext.GetNotificationTitle(),
+				"description": w.truncate(evalContext.Rule.Message, 512),
+				"url":         messageURL,
+				"picurl":      evalContext.ImagePublicURL,
+			},
+		},
+	}
+	return json.Marshal(body)
+}
+
+func (w *WeComNotifier) buildMarkdownBody(evalContext *alerting.EvalContext) ([]byte, error) {
```

Comment:
If possible, can you post a screenshot of how this looks like on the platform? Thank you!

Reply:
Of course, it looks like below: 

Test Notification message:
![image](https://user-images.githubusercontent.com/10809900/142375037-1cc53508-eb8e-40bb-bcb7-b9a1f32bca5c.png)

real-world message with some prometheus metrics:

![image](https://user-images.githubusercontent.com/10809900/142375289-5f9b5e60-29fb-40cf-b81b-1d6697dec32e.png)